src/DynamoCoreWpf/Utilities/CrashReportTool.cs

```diff
@@ -0,0 +1,228 @@
+﻿using Dynamo.Core;
+using Dynamo.Models;
+using Dynamo.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Dynamo.Core
+{
+    internal class CrashReportArgs : EventArgs
+    {
+        public bool SendLogFile = true;
+        public bool SendSettingsFile = true;
+        public bool SendDynFile = true;
+        public bool SendRecordedCommands = true;
+
+        internal DynamoViewModel viewModel;
+
+        internal CrashReportArgs(DynamoViewModel dynamoViewModel)
+        {
+            viewModel = dynamoViewModel;
+        }
+    }
+
+    internal class CrashReportTool
+    {
+        static CrashReportTool()
+        {
+            try
+            {
+                var assemblyConfig = ConfigurationManager.OpenExeConfiguration(Assembly.GetExecutingAssembly().Location);
+                CerToolLocation = assemblyConfig?.AppSettings.Settings["CERLocation"]?.Value;
+            }
+            catch
+            {}
+        }
+
+        private static string CerToolLocation;
+
+        private enum MINIDUMP_TYPE
+        {
+            MiniDumpNormal = 0x00000000,
+            MiniDumpWithDataSegs = 0x00000001,
+            MiniDumpWithFullMemory = 0x00000002,
+            MiniDumpWithHandleData = 0x00000004,
+            MiniDumpFilterMemory = 0x00000008,
+            MiniDumpScanMemory = 0x00000010,
+            MiniDumpWithUnloadedModules = 0x00000020,
+            MiniDumpWithIndirectlyReferencedMemory = 0x00000040,
+            MiniDumpFilterModulePaths = 0x00000080,
+            MiniDumpWithProcessThreadData = 0x00000100,
+            MiniDumpWithPrivateReadWriteMemory = 0x00000200,
+            MiniDumpWithoutOptionalData = 0x00000400,
+            MiniDumpWithFullMemoryInfo = 0x00000800,
+            MiniDumpWithThreadInfo = 0x00001000,
+            MiniDumpWithCodeSegs = 0x00002000,
+            MiniDumpWithoutAuxiliaryState = 0x00004000,
+            MiniDumpWithFullAuxiliaryState = 0x00008000,
+            MiniDumpWithPrivateWriteCopyMemory = 0x00010000,
+            MiniDumpIgnoreInaccessibleMemory = 0x00020000,
+            MiniDumpWithTokenInformation = 0x00040000,
+            MiniDumpWithModuleHeaders = 0x00080000,
+            MiniDumpFilterTriage = 0x00100000,
+            MiniDumpWithAvxXStateContext = 0x00200000,
+            MiniDumpWithIptTrace = 0x00400000,
+            MiniDumpScanInaccessiblePartialPages = 0x00800000,
+            MiniDumpFilterWriteCombinedMemory,
+            MiniDumpValidTypeFlags = 0x01ffffff
+        };
+
+        [DllImport("kernel32.dll")]
+        private static extern uint GetCurrentThreadId();
+
+        [DllImport("dbghelp.dll")]
+        private static extern bool MiniDumpWriteDump(
```

Comment:
is this from a reference somewhere? Maybe link would be good?

Reply:
dbghelp.dll is part of windows.
https://docs.microsoft.com/en-us/windows/win32/debug/dbghelp-versions
The docs do say that older versions might exist, but I think `MiniDumpWriteDump` is a very old API and it is probably on all versions of the dll.
It is also available as a redistributable if we want to have it published with Dynamo (but I think it will bring in other dependencies too)